### ISSUE TITLE
ENH: info shows approx. mem size?

### ISSUE THREAD

**Issue by jona-sassenhagen** [mne-tools/mne-python#3401]
When you call a Pandas Dataframe's info method, it shows you an approximation of the memory usage:

```
df.info()
<class 'pandas.core.frame.DataFrame'>
MultiIndex: 17182 entries, (Vis/L, 0, -200) to (Vis/R, 241, 500)
Data columns (total 59 columns):
EEG 001    17182 non-null float64
EEG 002    17182 non-null float64
...
EEG 060    17182 non-null float64
dtypes: float64(59)
memory usage: 7.9+ MB
```

Would it in principle be possible to show something like this for MNE objects, for example, in the `__repr__`, the info's `__repr__`, or as a special private method?

Not a big thing, but might be cool for when you want to parallelize across data sets, or want to estimate if an operation will blow up your memoy.

**Comment by Eric89GXL** [comment]
That would be cool

**Comment by agramfort** [comment]
+1

**Comment by mmagnuski** [comment]
+1 from me too

**Comment by Eric89GXL** [comment]
You could probably get pretty far by basing the estimate off the logic used
by object_hash.

But that might be too slow. If it is, we can get a rougher estimate from
just don't .data or ._data, as the object dictates.

Are you on it? If not I can try soon.

**Comment by jona-sassenhagen** [comment]
>Are you on it? If not I can try soon.

I don't know how I'd even approach this :) And I have a huge backlog of other stuff (EEGLAB IO fixes, ERP plot, my open PRs etc).

**Comment by jona-sassenhagen** [comment]
Closed by #3408
